plugins/sh/grammar/sh.bnf

```diff
@@ -370,16 +370,19 @@ condition ::=
   | logical_or_condition
   | logical_and_condition
   | equality_condition
+  | regex_condition
   | comparison_condition
   | logical_bitwise_condition
   | literal_condition
   | parentheses_condition
 
 assignment_condition ::= condition '=' condition
-logical_or_condition ::= condition '||' condition
-logical_and_condition ::= condition '&&' condition
-equality_condition ::= condition ('==' | '!=' | '=~') condition
+logical_or_condition ::= condition newlines '||' newlines condition
+logical_and_condition ::= condition newlines '&&' newlines condition
+equality_condition ::= condition ('==' | '!=') condition
+regex_condition ::= condition '=~' regex_pattern
 comparison_condition ::= condition ('<' | '>') condition
 logical_bitwise_condition ::= '!' simple_command
 parentheses_condition ::= '(' condition ')'
-literal_condition ::= (w newlines | newlines w)
\ No newline at end of file
+literal_condition ::= (w newlines | newlines w)
+regex_pattern ::= w+
```

Comment:
Why not make the `regex_pattern` as private one to avoid redundant tree nodes

Reply:
I agree that the `regex_pattern` is redundant for single `w` tokens. If there's more than one element, e.g. as in `[[ "a b" =~ (a | b | c) ]]`, then I'd like to wrap them. Working with the PSI tree for this kind of structure would be quite difficult later, imo.
(A PSI element to wrap regex groups would be even better, but that's for a later improvement.)

I could try to update this to remove the `regex_pattern` for single `w` elements. Do you agree or would you prefer something different?

Reply:
If so, let's leave it as it is